ethpm_cli/install.py

```diff
@@ -47,11 +54,39 @@ def install_package(pkg: Package, config: Config) -> None:
     tmp_pkg_dir = Path(tempfile.mkdtemp())
     write_pkg_installation_files(pkg, tmp_pkg_dir, config.ipfs_backend)
 
-    # Copy temp pacakge directory to ethpm dir namespace
+    # Copy temp package directory to ethpm dir namespace
     dest_pkg_dir = config.ethpm_dir / pkg.alias
     validate_parent_directory(config.ethpm_dir, dest_pkg_dir)
     shutil.copytree(tmp_pkg_dir, dest_pkg_dir)
-    update_ethpm_lock(pkg, (config.ethpm_dir / "ethpm.lock"))
+    install_to_ethpm_lock(pkg, (config.ethpm_dir / "ethpm.lock"))
+
+
+def list_installed_packages(config: Config) -> None:
+    for pkg_data in get_installed_packages(config.ethpm_dir):
+        logger.info(pkg_data)
+
+
+@to_list
+def get_installed_packages(ethpm_dir: Path) -> Iterable[str]:
+    installed_pkgs = reversed(sorted(ethpm_dir.glob("**/manifest.json")))
+    for manifest_path in installed_pkgs:
+        manifest = json.loads(manifest_path.read_text())
+        num_deep = str(manifest_path.relative_to(ethpm_dir)).count("/")
+        yield f"{num_deep * '--'} <Package {manifest['package_name']}=={manifest['version']}>"
```

Comment:
While this is fine and doesn't absolutely need to be addressed, I think you will want to restructure these function a little.

- `get_installed_package_tree`: Returns a data structure which represents the full tree of installed packages.
- A mechanism for displaying that information.

Here's an idea that might work well.

```python
class InstalledPackageTree(NamedTuple):
    depth: int
    path: Path
    manifest: ...
    children: Tuple["InstalledPackageTree", ...]  # self referencing, mypy might not like this but data structure wise this should be fine.
```

You should be able to use this data structure to store the data about the package tree.

Next, a recursive implementation that generates the data structure.

For each of the top level dependencies, populate them using the following function.

```python
def get_installed_package_tree(base_dir, depth=0):
    manifest = ...  # the loaded manifest (either raw JSON or the parsed internal representation)
    dependency_dirs = ...  # the base directories for each of this packages installed dependencies
    children = tuple(
        get_installed_package_tree(dependency_dir, depth + 1)
        for dependency_dir in dependency_dirs
    )
    return InstalledPackageTee(depth, base_dir, manifest, children)
```

You should then have a tuple/iterable of `InstalledPackageTree` objects that represent all of the installed packages.

Next to display the data, we can just implement a method on the `InstalledPackageTree`.  This could be `__str__` or something more formal like `format_for_display()`.

```python
def __str__(self) -> str:
    prefix = "  " * depth
    if self.children:
        children = "\n" + "\n".join((str(child) for child in self.children))
    else:
        children = ""
    return (
        f"{prefix}{self.package_identifier}=={self.package_version} ({self.content_hash_as_hex}){children}"
    )
```

Then your function to display these is just to log the `__str__` representation of each of the top level `InstalledPackageTree` objects to the terminal.